evap/evaluation/models.py

```diff
@@ -164,8 +164,14 @@ class Visibility(models.IntegerChoices):
 
     visibility = models.IntegerField(choices=Visibility.choices, verbose_name=_('visibility'), default=Visibility.MANAGERS)
 
+    is_locked = models.BooleanField(verbose_name=_("is locked"), default=False)
+
     objects = QuestionnaireManager()
 
+    def clean(self):
+        if self.type == self.Type.CONTRIBUTOR and self.is_locked:
+            raise ValidationError({'is_locked': _('Contributor questionnaires cannot be locked.')})
```

Comment:
I decided against this, because the documentation says constraint violations would raise a database integrity error, which feels wrong. Also I'm not sure whether the error would be properly displayed in the form, with a nice error message.

Reply:
database integrity errors will be handled as 500 error messages and an email will be send to the admins.